src/azure-cli-core/azure/cli/core/application.py

```diff
@@ -44,6 +44,23 @@ def get_command_table(self, argv=None):  # pylint: disable=no-self-use
         matches = Configuration.find_matches(argv, command_tree)
         return dict(matches)
 
+    # raise customized exceptions on removed commands if the command module like
+    @staticmethod
+    def err_on_removed_command(argv):
```

Comment:
See comments below - if we register a generic "this command has been removed' handler for the removed command, there is no need to remove things from argv/fail early. As it is now, you are executing this code when someone does tab completion (which I believe will still work, based on the current version of argcomplete), but it isn't what you would expect to have happen...